scripts/in_container/run_generate_constraints.py

```diff
@@ -304,76 +312,21 @@ def generate_constraints_pypi_providers(config_params: ConfigParams) -> None:
-                console.print(
-                    f"[yellow]Installing {file.name} from local dist folder as it is a chicken egg provider"
-                )
-                packages_to_install.append(f"{provider_package} @ file://{file.as_posix()}")
-                break
-            else:
-                console.print(
-                    f"[yellow]Skipping {provider_package} as it is not found in dist folder to install."
-                )
-            # Skip checking if chicken egg provider is available in PyPI - it does not have to be there
-            continue
-        console.print(f"[bright_blue]Checking if {provider_package} is available in PyPI: ... ", end="")
-        r = requests.get(f"https://pypi.org/pypi/{provider_package}/json", timeout=60)
-        if r.status_code == 200:
-            version = r.json()["info"]["version"]
-            console.print("[green]OK")
-            # TODO: In the future we might make it a bit more sophisticated and allow to install
-            # earlier versions of providers here - but for now we just install the latest version
-            # But in practice latest version does not have to be supported by current version of airflow
-            # This should be a very rare case though - such provider distributions should only be released
-            # in pre-release mode
-            packages_to_install.append(f"{provider_package}=={version}")
-        else:
-            console.print("[yellow]NOK. Skipping.")
-    find_airflow_distributions = AIRFLOW_DIST_PATH.glob("apache_airflow-*.whl")
-    airflow_install = "."
-    if find_airflow_distributions:
-        airflow_install = next(find_airflow_distributions).as_posix()
-    airflow_core_install = "./airflow-core[all]"
-    find_airflow_core_distributions = AIRFLOW_DIST_PATH.glob("apache_airflow_core-*.whl")
-    if find_airflow_core_distributions:
-        airflow_core_install = next(find_airflow_core_distributions).as_posix() + "[all]"
-    airflow_task_sdk_install = "./airflow-task-sdk"
-    find_airflow_task_sdk_distribution = AIRFLOW_DIST_PATH.glob("apache_airflow_task_sdk-*.whl")
-    if find_airflow_task_sdk_distribution:
-        airflow_task_sdk_install = next(find_airflow_task_sdk_distribution).as_posix()
     run_command(
         cmd=[
             "uv",
             "pip",
             "install",
             "--no-sources",
-            airflow_install,
-            airflow_core_install,
-            airflow_task_sdk_install,
+            "apache-airflow[all]",
+            "apache-airflow-core[all]",
+            "apache-airflow-task-sdk",
             "./airflow-ctl",
-            "--reinstall",  # We need to pull the provider distributions from PyPI - not use the local ones
-            *packages_to_install,
+            "--reinstall",  # We need to pull the provider distributions from PyPI or dist, not the local ones
             "--resolution",
             "highest",
+            "--find-links",
```

Comment:
We can now simply rely on `uv pip` finding  newer versions of providers in the `dist` folder where we prepared all the non-released providers yet. Previously we had to decide which packages we build and manually choose those packages because we were not sure if they were released yet or not.